test/test.js

```diff
@@ -0,0 +1,298 @@
+const container = require("../index.js").container;
+const assert = require("assert");
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
+
+describe("dependency injection", function() {
+  let subject;
+
+  beforeEach(function() {
+    subject = container();
+  });
+
+  describe("#register", function() {
+    it("should register a factory", function() {
+      subject.register("logger", function() {
+        return "message";
+      });
+      assert.equal(subject.get("logger"), "message");
+    });
+
+    it("should register an object", function() {
+      subject.register("logger", "message");
```

Comment:
I'm on the fence about this feature. Especially in ES6, the difference in code seems trivial (e.g. `() => "message"`). That would require being able to parse arrow functions, though.

Reply:
Well, we had also batted around the idea of a new api similar to what angular 1 did, `.constant(name, string)`. So maybe it's better to codify this functionality that way? I do sort of agree with what you're saying tho.

Reply:
Depending on the general consensus of `.constant`, I have a working version with tests ready to go.

Reply:
For this first version, I'd rather focus on making the API _smaller_ and better-tested. Once we have confidence in the core, we can branch out into different improvements. (For example, can we get away from function parsing? How do we make it easier to register _functions_ as dependencies?)

Reply:
Well, we could pull in esprima and do some AST parsing to see if that yields an alternative approach, but then we're a bit heavier on the dependency side.

Reply:
Also, for additional features we had talked about the idea of a plugin architecture, perhaps `dependable-test`, and `dependable-ext` for additional features? That way we can grow features without adding bloat to the core, well-tested, library.